### PR TITLE
perf: enable inductor combo_kernels for horizontal fusion

### PR DESCRIPTION
**Pull request by jasperjiaguo**
Enable `combo_kernels` and `benchmark_combo_kernel` in inductor config to allow horizontal fusion of sibling ops with different shapes. This fuses operations like q_norm + k_norm (QK normalization) into a single triton kernel instead of generating separate kernels for each.

Requires torch >= 2.9.0.
Screenshot 2026-04-02 at 3 49 15 PM
Screenshot 2026-04-02 at 3 47 59 PM

## Profile Results

Qwen3-0.6B FP8 embeddings on H200, PCG inductor, 7k tokens:

| Metric | Before | After |
|--------|:------:|:-----:|
| GPU kernels per forward | 413 | **357** (-14%) |
| QK norm kernels per layer | 4 | **2** |
| `split_with_sizes` / `clone` in kernel names | Present | **Gone** |

The QK norm reduction + pointwise kernels for q and k are now horizontally fused into single kernels.

Throughput impact is neutral at 60 RPS (kernel launch overhead is not the bottleneck at this load), but the reduced kernel count should help at higher concurrency or with smaller models where launch overhead is proportionally larger.

### CODE REVIEW

**Comment by jasperjiaguo** [comment]
Yes, this requires `--piecewise-cuda-graph-compiler inductor`. The combo_kernels config is set during inductor compilation in `piecewise_cuda_graph_runner.py`.

Reproduction command (tested with Qwen3-0.6B FP8 on H200):
```bash
python -m sglang.launch_server \
  --model-path Qwen/Qwen3-0.6B-FP8 \
  --tp 1 --dp 1 --dtype float16 \
  --is-embedding \
  --disable-radix-cache \
  --chunked-prefill-size -1 \
  --max-prefill-tokens 32768 \
  --context-length 32768 \
  --enable-piecewise-cuda-graph \
  --piecewise-cuda-graph-max-tokens 32768 \
  --piecewise-cuda-graph-compiler inductor \
  --port 30000
```

The `combo_kernels` config enables horizontal fusion of sibling ops (e.g., q_norm + k_norm into a single triton kernel). In our testing it reduced GPU kernel count from 413 to 357 per forward pass.

**Comment by Oasis-Git** [comment]
Sounds good. I think the function is good to merge but the implementation should be fixed:
1. I think we can put the change under compilation folder instead of Piecewise Cuda Graph python file? I think config.py might be good.
2. Maybe we can use some conditions to control the enable: for example, if `compile backend = inductor`. The benefit here is that we can bypass most of the ci which I noticed that you spent much time on and not disturb the default eager mode.

**Comment by jasperjiaguo** [comment]
Thanks for the review! Addressed both suggestions:

1. Moved `combo_kernels` config from `piecewise_cuda_graph_runner.py` to `compilation/backend.py` (`SGLangBackend.__init__`)
2. Gated on `config.piecewise_cuda_graph_compiler == "inductor"` so it only activates with inductor backend, not eager mode

Also removed the now-unused `import torch._inductor.config` from the runner.

**Review by Oasis-Git — APPROVED** [review]
